zerver/tests/test_push_notifications.py

```diff
@@ -1342,6 +1345,148 @@ def test_push_bouncer_api(self) -> None:
         self.assertEqual(remote_realm.last_request_datetime, time_sent)
 
 
+class RegisterPushDeviceToBouncer(BouncerTestCase):
+    DEFAULT_SUBDOMAIN = ""
+
+    def get_register_push_device_payload(
+        self,
+        token: str = "apple-tokenaz",
+        token_kind: int = RemotePushDevice.TokenKind.APNS,
+        ios_app_id: str | None = "example.app",
+        timestamp: int = datetime_to_timestamp(now()),
```

Comment:
Python defaults are evaluated at function definition time, so this gives the timestamp at which the function was defined, not the timestamp at which it was called.

Reply:
If code that looks like it's buggy is intentional, it HAS to have a comment explaining what you're doing. Code that looks buggy is very expensive for people reading code, both during code review and for years into the future.

But once you're doing that, you almost always will decide it's better to write it the way that doesn't look like a bug.

Reply:
```diff
-        timestamp: int = datetime_to_timestamp(now()),
+        timestamp: int | None = None,
     ) -> dict[str, str | int]:
         hamlet = self.example_user("hamlet")
         remote_realm = RemoteRealm.objects.get(uuid=hamlet.realm.uuid)
 
+        if timestamp is None:
+            timestamp = datetime_to_timestamp(now())
+
```